src/Parser/Monk/Brewmaster/Modules/Spells/AzeriteTraits/TrainingOfNiuzao.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { calculateAzeriteEffects } from 'common/stats';
+import { formatNumber, formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+const TON_SCALE = {
+  [SPELLS.LIGHT_STAGGER_DEBUFF.id]: 1/3,
+  [SPELLS.MODERATE_STAGGER_DEBUFF.id]: 2/3,
+  [SPELLS.HEAVY_STAGGER_DEBUFF.id]: 1,
+};
+
+export function trainingOfNiuzao_stats(combatant) {
+  if(!combatant.hasTrait(SPELLS.TRAINING_OF_NIUZAO.id)) {
+    return null;
+  }
+  let mastery = 0;
+  for(const rank of combatant.traitsBySpellId[SPELLS.TRAINING_OF_NIUZAO.id]) {
```

Comment:
Not too keen on the choice of a for loop here over a reducer, but it's not a blocker.